### PR TITLE
Change download_golang script for RPM Build download path

### PR DESCRIPTION
**Pull request by zafs23**
*Issue #, if available:*

*Description of changes:* This PR is the second part of [this PR](https://github.com/aws/eks-distro-build-tooling/pull/808). The version.yml file has also updated with the latest golang versions as per of golang new [release PR](https://github.com/aws/eks-distro-build-tooling/pull/811).

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: builder-base/versions.yaml [hunk 1]
```diff
@@ -3,11 +3,12 @@ ANSIBLE_VERSION: 2.10.0
...
-GOLANG_VERSION_116: 1.16.15-3
-GOLANG_VERSION_117: 1.17.13-2
-GOLANG_VERSION_118: 1.18.10-1
-GOLANG_VERSION_119: 1.19.5-1
+GOLANG_VERSION_115: 1.15.15-3
+GOLANG_VERSION_116: 1.16.15-4
+GOLANG_VERSION_117: 1.17.13-3
+GOLANG_VERSION_118: 1.18.10-2
+GOLANG_VERSION_119: 1.19.5-2
+GOLANG_VERSION_120: 1.20.0-1
```
Comment (danbudris) [inline builder-base/versions.yaml:11]: not for this PR, but just an FYI, we'll want to update the Builder Base to use this version as well -- there are instructions in the runbook for a new go version addition that we'll have to follow

File: builder-base/Dockerfile [hunk 2]
```diff
@@ -406,6 +436,8 @@ COPY --link --from=golang-1.17 /golang-1.17 /
 COPY --link --from=go-licenses-1.17 /go-licenses-1.17 /
 COPY --link --from=golang-1.19 /golang-1.19 /
 COPY --link --from=go-licenses-1.19 /go-licenses-1.19 /
+COPY --link --from=golang-1.20 /golang-1.20 /
+COPY --link --from=go-licenses-1.20 /go-licenses-1.20 /
```
Comment (zafs23) [inline builder-base/Dockerfile:440]: @danbudris not sure if I should add go-licenses-1.20 here or not

File: eks-distro-base/Makefile [hunk 3]
```diff
@@ -202,11 +202,12 @@ BASE_NODEJS_COMPILER_VARIANT_VERSIONS=$(foreach ver,$(BASE_COMPILER_VARIANT_VERS
...
-GOLANG_1.16_FULL_VERSION=1.16.15-3
-GOLANG_1.17_FULL_VERSION=1.17.13-2
-GOLANG_1.18_FULL_VERSION=1.18.10-1
-GOLANG_1.19_FULL_VERSION=1.19.5-1
+GOLANG_1.15_FULL_VERSION=1.15.15-3
+GOLANG_1.16_FULL_VERSION=1.16.15-4
+GOLANG_1.17_FULL_VERSION=1.17.13-3
+GOLANG_1.18_FULL_VERSION=1.18.10-2
+GOLANG_1.19_FULL_VERSION=1.19.5-2
+GOLANG_1.19_FULL_VERSION=1.20.0-1
```
Comment (danbudris) [inline eks-distro-base/Makefile:210]:
```suggestion
GOLANG_1.20_FULL_VERSION=1.20.0-1
```

**Comment by danbudris** [comment]
it looks like the `1.20` artifacts are being uploaded to the S3 bucket with the `.0` path version truncated -- and up as `1.20` rather than `1.20.0`

File: builder-base/versions.yaml (hunk 1 above)
Comment (danbudris) [inline builder-base/versions.yaml:11]:
```suggestion
GOLANG_VERSION_120: 1.20-1
```

File: eks-distro-base/Makefile [hunk 4]
```diff
@@ -200,13 +200,14 @@ BASE_NODEJS_COMPILER_VARIANT_VERSIONS=$(foreach ver,$(BASE_COMPILER_VARIANT_VERS
...
-GOLANG_1.18_FULL_VERSION=1.18.10-1
-GOLANG_1.19_FULL_VERSION=1.19.5-1
+BASE_GOLANG_VARIANT_VERSIONS=1.15 1.16 1.17 1.18 1.19 1.20
+
+GOLANG_1.15_FULL_VERSION=1.15.15-3
+GOLANG_1.16_FULL_VERSION=1.16.15-4
+GOLANG_1.17_FULL_VERSION=1.17.13-3
+GOLANG_1.18_FULL_VERSION=1.18.10-2
+GOLANG_1.19_FULL_VERSION=1.19.5-2
+GOLANG_1.20_FULL_VERSION=1.20.0-1
```
Comment (danbudris) [inline eks-distro-base/Makefile:210]:
```suggestion
GOLANG_1.20_FULL_VERSION=1.20-1
```

**Pull request merged**